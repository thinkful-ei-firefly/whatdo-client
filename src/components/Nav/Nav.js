import React from 'react'
import { Link } from 'react-router-dom'
import './Nav.css'

export default class Nav extends React.Component {

/*
static contextType = UserContext

  handleLogoutClick = () => {
    this.context.processLogout()
  }

  renderLogoutLink() {
    return (
      <>
          <Link
            onClick={this.handleLogoutClick}
            to='/Login'
            className="btn btn2">
            Logout
          </Link>
      </>
    )
  }

  renderLoginLink() {
    return (
      <>
        <Link to='/signIn' className="btn btn2">Login</Link>
        <Link to='/signUp' className="btn btn2">Sign up</Link>
      </>
    )
  }
*/

  
  render() {
    return (
      <nav className='login-controls'  role='navigation'>
      <input type="checkbox" id="chk"></input>
        <label htmlFor="chk" className="show-menu-btn">
          <i className="fas fa-ellipsis-h"></i>
        </label>
    
        <ul className="menu">

        <Link to='/eventsPage' className="btn btn2">Events</Link>

        <Link to='/myevents' className="btn btn2">My Event</Link>

        {/** 
        TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()
        */}
        {/**demo  login & register*/}
        <Link to='/signIn' className="btn btn2">Sign in</Link>
        <Link to='/signUp' className="btn btn2">Sign up</Link>


        <label htmlFor="chk" className="hide-menu-btn">
            <i className="fas fa-times"></i>
          </label>
        </ul>
      </nav>
    )
  }
}
