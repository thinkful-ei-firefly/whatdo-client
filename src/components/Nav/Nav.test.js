import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Nav from './Nav'
import renderer from 'react-test-renderer'

describe('<Nav />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
