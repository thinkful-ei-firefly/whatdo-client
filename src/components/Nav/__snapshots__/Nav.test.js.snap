// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders as expected 1`] = `
<nav
  className="header__nav"
  role="navigation"
>
  <input
    id="chk"
    type="checkbox"
  />
  <label
    className="show-menu-btn"
    htmlFor="chk"
  >
    <i
      className="fas fa-ellipsis-h"
    />
  </label>
  <ul
    className="menu"
  >
    <a
      className="btn btn2"
      href="/eventsPage"
      onClick={[Function]}
    >
      Events
    </a>
    <a
      className="btn btn2"
      href="/myEventsPage"
      onClick={[Function]}
    >
      My Event
    </a>
    <a
      className="btn btn2"
      href="/signIn"
      onClick={[Function]}
    >
      Sign in
    </a>
     
    <a
      className="btn btn2"
      href="/signUp"
      onClick={[Function]}
    >
      Sign up
    </a>
    <label
      className="hide-menu-btn"
      htmlFor="chk"
    >
      <i
        className="fas fa-times"
      />
    </label>
  </ul>
</nav>
`;
